### PR TITLE
Upgrade dependencies to dcrd v2.0.2

### LINKED ISSUE #560: upgrade to dcrd v2.0

### PR DESCRIPTION
**Pull request by dmigwi**
closes #560 

<strong>NOTE TO TESTERS: Testing this PR, upgrades your database which can no longer be downgraded.</strong>

This PR upgrades to `decred.org/dcrwallet/v4` that depends on `github.com/decred/dcrd v2.0.2`

More changes introduced as a result of the dependencies upgrade.:
- Replace the client-server with peer-to-peer coin shuffle mixing
- Replaces the deprecated AgendaChoice struct with the map[string]string]
- Reuse the upstream VSPTicket methods introduced
- Remove AddressReuse from wallet config
- Upgrades the golang version to go 1.20 - (This was to fix some CI error documented here: https://github.com/golangci/golangci-lint-action/issues/552#issuecomment-2102806360)
- Downgrades the CI environment from `ubuntu-latest` to `ubuntu-20.04` due to  CI linter error documented here: https://github.com/actions/runner-images/issues/6709#issuecomment-1361269639
- DcrDex upgrade not possible for now because its ltc dependencies are buggy.

**Tests Run on the code.**
- [x] Coinshuffle mixing using the peer-to-peer architecture.
- [x] Test buying tickets
- [ ] Test voting functionality using live tickets

### CODE REVIEW

**Comment by dmigwi** [comment]
Coins Mixing using the new implementation works.
Screenshot 2024-06-19 at 20 09 30

**Comment by dmigwi** [comment]
The failure of the Linter tool has been resolved here: https://github.com/crypto-power/cryptopower/pull/545

**Comment by dreacot** [comment]
fix conflicts

File: libwallet/assets/dcr/dex_wallet.go [hunk 1]
```diff
@@ -18,9 +18,15 @@ import (
 	dexasset "decred.org/dcrdex/client/asset"
 	dexdcr "decred.org/dcrdex/client/asset/dcr"
 	"decred.org/dcrdex/dex"
-	walleterrors "decred.org/dcrwallet/v3/errors"
+
+	// Version v3 of dcrwallet library is still in use because dcrdex is yet to upgrade
+	// to dcrd v2.
```
Comment (dreacot) [inline libwallet/assets/dcr/dex_wallet.go:23]: you mean v4?

**Review by dreacot — COMMENTED** [review]
ACK

**Comment by dreacot** [comment]
NOTE TO TESTERS: Testing this PR upgrades your database which can no longer be downgraded after

File: libwallet/assets/dcr/dex_wallet.go (hunk 1 above)
Comment (dmigwi) [reply libwallet/assets/dcr/dex_wallet.go:23]: v4 is the tag but v2.0.2 is the release version. I will use v4 instead.

**Comment by dmigwi** [comment]
> NOTE TO TESTERS: Testing this PR upgrades your database which can no longer be downgraded after

I noticed that! Sorry for not sharing that information earliest possible.

**Comment by dmigwi** [comment]
I have identified the following error while testing the ticket purchase functionality: Fixing it ASAP:

```
2024-06-24 15:31:42.389 [ERR] DCR: AccountMixer instance errored: context canceled
2024-06-24 15:31:44.182 [INF] DCR: Published ticket purchase 67f0abc0ea1fe8b38929d0505da81005d658156dca4367b653b1ed579e344e01
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10278a97a]

goroutine 27445 [running]:
decred.org/dcrwallet/v4/wallet/udb.(*Store).Tx(0xc003644000, {0x105f4d148, 0xc004aa0200}, 0x0)
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/udb/txquery.go:262 +0x3a
decred.org/dcrwallet/v4/wallet.(*Wallet).GetTransactionsByHashes.func1({0x10410c9e0?, 0xc0069d2a48?})
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/wallet.go:2337 +0xe6
decred.org/dcrwallet/v4/wallet/walletdb.View({0x104110018, 0x105076380}, {0x104112a28, 0xc0032bc488}, 0xc003fb0e00)
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/walletdb/interface.go:222 +0x1a6
decred.org/dcrwallet/v4/wallet.(*Wallet).GetTransactionsByHashes(0xc0032bc488?, {0x104110018?, 0x105076380?}, {0xc003fb0ef8?, 0x1028a51c0?, 0xc003fb0e60?})
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/wallet.go:2334 +0xd2
decred.org/dcrwallet/v4/wallet.(*Wallet).NewVSPTicket(0xc0032bc6c8, {0x104110018, 0x105076380}, 0x0)
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/vspticket.go:43 +0x65
github.com/crypto-power/cryptopower/libwallet/internal/vsp.(*feePayment).receiveFeeAddress(0xc0045b2140)
	/Users/migwindungu/theSecretCoder/App/Golang/src/github.com/dmigwi/cryptopower/libwallet/internal/vsp/feepayment.go:275 +0xa5
github.com/crypto-power/cryptopower/libwallet/internal/vsp.(*Client).Process(0xc0052672c0, {0x1041115d0, 0xc006024600}, 0xc0066de7e0, 0xc007e36230, {0x10411a6e8?, 0x1d0320?, 0xc0?})
	/Users/migwindungu/theSecretCoder/App/Golang/src/github.com/dmigwi/cryptopower/libwallet/internal/vsp/vsp.go:252 +0x259
github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).buyTicket.func1({0x1041115d0?, 0xc006024600?}, 0xc006024600?, 0xc0076cadc0?)
	/Users/migwindungu/theSecretCoder/App/Golang/src/github.com/dmigwi/cryptopower/libwallet/assets/dcr/ticket.go:477 +0x45
decred.org/dcrwallet/v4/wallet.(*Wallet).purchaseTickets(0xc0032bc6c8, {0x1041115d0, 0xc006024600}, {0x1035197fe, 0x16}, {0x10411a638, 0xc003138ea0}, 0xc003fb1ec0)
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/createtx.go:1874 +0x3098
decred.org/dcrwallet/v4/wallet.(*Wallet).PurchaseTickets(0xc0032bc6c8, {0x1041115d0, 0xc006024600}, {0x10411a638, 0xc003138ea0}, 0xc003fb1ec0)
	/Users/migwindungu/theSecretCoder/App/Golang/pkg/mod/decred.org/dcrwallet/v4@v4.1.1/wallet/wallet.go:1611 +0x67
github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).buyTicket(0xc0031085b0, {0x104111608?, 0xc0054534f0?}, {0x104e78d70, 0x1}, 0x199bdfe32, 0x159600, 0xc0057a80f0)
	/Users/migwindungu/theSecretCoder/App/Golang/src/github.com/dmigwi/cryptopower/libwallet/assets/dcr/ticket.go:489 +0x2db
github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).runTicketBuyer.func1()
	/Users/migwindungu/theSecretCoder/App/Golang/src/github.com/dmigwi/cryptopower/libwallet/assets/dcr/ticket.go:417 +0x5a
created by github.com/crypto-power/cryptopower/libwallet/assets/dcr.(*Asset).runTicketBuyer in goroutine 27459
	/Users/migwindungu/theSecretCoder/App/Golang/src/github.com/dmigwi/cryptopower/libwallet/assets/dcr/ticket.go:439 +0xb2b
```

**Comment by dreacot** [comment]
The confirm password modal to start mixing is stuck spinning

Screenshot from 2024-06-26 09-08-49

The disconnection/re-connection modal keeps stacking themselves, i have to "infinitely" keep clicking okay to close each of them with no end in sight

Screenshot from 2024-06-25 17-08-04

**Comment by dreacot** [comment]
i see this error, but at no point am i asked to unlock my wallet

```
2024-06-26 09:05:00.740 [WRN] DCR: unable to getWallet info using ticket: c55f8dee47ae72b29b6b27f36eec6b05219f2875d3dfd0ef06f13b52c78c7093 Error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-06-26 09:05:00.740 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-06-26 09:05:00.740 [WRN] DCR: unable to getWallet info using ticket: c55f8dee47ae72b29b6b27f36eec6b05219f2875d3dfd0ef06f13b52c78c7093 Error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-06-26 09:05:00.740 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
```

**Comment by dmigwi** [comment]
Confirmation that a ticket can be bought now with crashing the app or creating Tickets that are mined:

```
2024-06-29 18:36:20.819 [INF] DCR: [1] Running ticket buyer
2024-06-29 18:36:59.532 [INF] DCR: Reserved credits for 1 tickets: total fee: 0.00013512 DCR
2024-06-29 18:36:59.533 [INF] DCR: Returning address (account=2 branch=1 child=11)
2024-06-29 18:36:59.533 [INF] DCR: selected input d0846eabe98532a020d7f237dd1840484c5722cdcca0cd3c6b667dc8f3318440:14 (2.68435456 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.533 [INF] DCR: selected input 4ad647d23addd53ae30d9c0417bdc9a37f59b18d6b5a81835a69c976fab1a31b:8 (10.73741824 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.533 [INF] DCR: selected input a073d3d4d223642dd9fce58fa3d93143bc1e5ea43165a0eb0f37500e3d1d8632:0 (0.00262144 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.534 [INF] DCR: selected input a073d3d4d223642dd9fce58fa3d93143bc1e5ea43165a0eb0f37500e3d1d8632:2 (0.00262144 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.534 [INF] DCR: selected input 219272c54466e7422f36c8320071f020477d0c516a6c73513aad8c0ee01adaec:26 (2.68435456 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.534 [INF] DCR: selected input 4ad647d23addd53ae30d9c0417bdc9a37f59b18d6b5a81835a69c976fab1a31b:13 (10.73741824 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.534 [INF] DCR: selected input 219272c54466e7422f36c8320071f020477d0c516a6c73513aad8c0ee01adaec:17 (2.68435456 DCR) for ticket purchase split transaction
2024-06-29 18:36:59.534 [INF] DCR: selected input 14f67030a45fc61fcd4070f083eca700b8e422e5f7c81f602a45127074e94b27:12 (68.88778044 DCR) for ticket purchase split transaction
2024-06-29 18:39:00.076 [INF] DCR: Returning address (account=1 branch=0 child=26)
2024-06-29 18:39:43.707 [INF] WLLT: Inserting unconfirmed transaction 5be9afc97b6499a6db61cd7d4c98d174e742a8e9c1b9ab7eb7befeec801d0e39
2024-06-29 18:39:43.772 [INF] DCR: Completed CoinShuffle++ mix of ticket split transaction 5be9afc97b6499a6db61cd7d4c98d174e742a8e9c1b9ab7eb7befeec801d0e39
2024-06-29 18:39:43.786 [INF] DCR: [1] New Transaction 5be9afc97b6499a6db61cd7d4c98d174e742a8e9c1b9ab7eb7befeec801d0e39
2024-06-29 18:39:43.884 [INF] DCR: Split output is 5be9afc97b6499a6db61cd7d4c98d174e742a8e9c1b9ab7eb7befeec801d0e39:7
2024-06-29 18:39:43.951 [INF] DCR: Returning address (account=1 branch=1 child=0)
2024-06-29 18:39:43.995 [INF] DCR: Returning address (account=1 branch=0 child=27)
2024-06-29 18:39:43.996 [INF] WLLT: Inserting unconfirmed transaction 869f276d35aa6009bba99d087bbf58b915b2cc06683f73e7486b8aaa04117584
2024-06-29 18:39:44.042 [INF] DCR: [1] New Transaction 869f276d35aa6009bba99d087bbf58b915b2cc06683f73e7486b8aaa04117584
2024-06-29 18:40:26.289 [INF] DCR: Published ticket purchase 869f276d35aa6009bba99d087bbf58b915b2cc06683f73e7486b8aaa04117584
2024-06-29 18:40:27.381 [INF] DCR: VSP requires fee 0.00012146 DCR
2024-06-29 18:40:27.420 [INF] DCR: Returning address (account=1 branch=1 child=1)
2024-06-29 18:40:27.455 [INF] WLLT: Inserting unconfirmed transaction 0293b0f8a9084e332fce2905b96ee247143610ff16c48c4b2c1e918366902e10
2024-06-29 18:40:27.588 [INF] DCR: [1] New Transaction 0293b0f8a9084e332fce2905b96ee247143610ff16c48c4b2c1e918366902e10
2024-06-29 18:40:27.727 [INF] DCR: successfully processed 869f276d35aa6009bba99d087bbf58b915b2cc06683f73e7486b8aaa04117584
2024-06-29 18:40:27.729 [INF] DCR: [1] Purchased ticket 869f276d35aa6009bba99d087bbf58b915b2cc06683f73e7486b8aaa04117584 at stake difficulty 68.58200368 DCR
2024-06-29 18:45:22.016 [ERR] UI: the USD exchange rate is disabled
2024-06-29 18:45:22.018 [ERR] UI: the USD exchange rate is disabled
```

**Comment by dmigwi** [comment]
@dreacot I don't understand why your confirm spinner reference above has been stuck. I think the issue might be related to your network because all along I have experienced such a  situation.

`The disconnection/re-connection modal keeps stacking themselves` I can't replicate this on my end please confirm if the issues still persists?

**Comment by dmigwi** [comment]
Newly bought ticket is now live:

Screenshot 2024-06-29 at 19 32 09

**Comment by dreacot** [comment]
please resolve conflicts

**Comment by dreacot** [comment]
i don't think "coordination server" is needed anymore @dmigwi 

Screenshot from 2024-07-01 12-41-34

**Comment by dreacot** [comment]
Mixing and Ticket buying confirmed working

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
@dreacot @dmigwi 
Do we need stakeShuffle before buy ticket on staking ? If so, notice users when they buy tickets without stakeshuffle

I get this error when buy ticket

```
2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
```

Also I get error when click to my ticket for show detail

```
2024-07-02 10:31:02.391 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-02 10:31:02.391 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-02 10:31:02.391 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-02 10:31:02.391 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
```

**Review by JoeGruffins — APPROVED** [review]

File: libwallet/assets/dcr/ticket.go [hunk 2]
```diff
@@ -196,15 +193,19 @@ func (asset *Asset) VSPTicketInfo(hash string) (*VSPTicketInfo, error) {
 		return ticketInfo, nil
 	}
 
-	vspClient, err := asset.VSPClient(walletTicketInfo.Host, walletTicketInfo.PubKey)
+	// Account being set to -1 means the client isn't being used to purchase
+	// tickets as the account policy configuration won't be set.
+	vspClient, err := asset.VSPClient(-1, walletTicketInfo.Host, walletTicketInfo.PubKey)
```
[the remark this answers is not in the document]
Comment (dmigwi) [reply libwallet/assets/dcr/ticket.go:198]: Some vsp client instances don't readily have the purchase account accessible and their use on those instances will not require the use of an account.
I will look into ways to always have an account set either to the default purchase account of the provided user account.

File: libwallet/assets/dcr/ticket.go [hunk 3]
```diff
@@ -238,49 +241,49 @@ func (asset *Asset) StartTicketBuyer(passphrase string) error {
...
-		cfg.VspClient, err = asset.VSPClient(cfg.VspHost, vspInfo.PubKey)
-	}
 	if err != nil {
 		return fmt.Errorf("error setting up vsp client: %v", err)
 	}
 
+	cfg.VspClient, err = asset.VSPClient(cfg.PurchaseAccount, cfg.VspHost, vspInfo.PubKey)
+	if err != nil {
+		log.Errorf("[%d] VSP Client instance failed error: %v", asset.ID, err)
+		return errors.New("VSP Client failed to start due to incorrect configuration")
```
[the remark this answers is not in the document]
Comment (dmigwi) [reply libwallet/assets/dcr/ticket.go:270]: The raw error from the vspclient maybe confusing or too long for the UI display in full. I experienced an instance where the upstream error returned was "internal error", this doesn't make sense to display to the user.

File: libwallet/assets/dcr/ticket.go [hunk 4]
```diff
@@ -449,34 +452,38 @@ func (asset *Asset) buyTicket(ctx context.Context, passphrase string, sdiff dcru
 		return err
 	}
 
+	if !asset.IsTicketBuyerAccountSet() {
+		return errors.New("Ticket purchase account is not set")
```
[the remark this answers is not in the document]
Comment (dmigwi) [reply libwallet/assets/dcr/ticket.go:456]: Okay noted.

**Review by dreacot — APPROVED** [review]

**Comment by JustinBeBoy** [comment]
> > @dreacot @dmigwi Do we need stakeShuffle before buy ticket on staking ? If so, notice users when they buy tickets without stakeshuffle
> > I get this error when buy ticket
> > ```
> > 2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.601 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > 2024-07-02 10:33:40.602 [ERR] DCR: [5] Ticket purchasing failed: Mixing and Unmixing Accounts are not set
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also I get error when click to my ticket for show detail
> > ```
> > 2024-07-02 10:31:02.391 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
> > 2024-07-02 10:31:02.391 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
> > 2024-07-02 10:31:02.391 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
> > 2024-07-02 10:31:02.391 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
> > ```
> 
> @JustinBeBoy Yes! The we need those account set up before the auto ticket purchasing functionality can run. For the ticket details error, It means that ticket wasn't successfully purchase. Please share more logs to help address the issue.

1. About running ticket buyer, please notice for user if it not set up it before
2.  About detail ticket: It's all log i have, I will show when you click into the ticket to go to ticket detail(every ticket include Revoke and voted).
This is new log when you update your code:
```
2024-07-04 14:37:00.704 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-07-04 14:37:00.704 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-07-04 14:38:09.207 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-07-04 14:38:09.207 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-07-04 14:38:17.897 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
2024-07-04 14:38:17.897 [ERR] UI: VSPTicketInfo error: wallet.DumpWIFPrivateKey: wallet locked:: wallet.LoadPrivateKey: default
```

**Comment by dmigwi** [comment]
@JustinBeBoy Your two issue raised above have been addressed. Please review the code again!

**Comment by JustinBeBoy** [comment]
@dmigwi 

I still get this log when click on my Revoked ticket(not watch only wallet, you need your wallet have ticket):
```
2024-07-05 13:58:13.607 [INF] DCR: [5] Running ticket buyer
2024-07-05 13:58:40.884 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-05 13:58:40.884 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-05 13:58:40.884 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-05 13:58:40.884 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
```

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
@dmigwi 

I still get this log when click on my Revoked ticket(not watch only wallet, you need your wallet have ticket):
```
2024-07-05 13:58:13.607 [INF] DCR: [5] Running ticket buyer
2024-07-05 13:58:40.884 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-05 13:58:40.884 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-05 13:58:40.884 [WRN] DCR: unable to getWallet info using ticket: cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384 Error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
2024-07-05 13:58:40.884 [ERR] UI: VSPTicketInfo error: item does not exist: no VSP info for ticket cac52a6d1c83d9688ef5f64aade9635dfb4918f4250a16969b5ad18f35a15384
```

**Comment by dmigwi** [comment]
> unable to getWallet info using ticket

Your revoked ticket above indicates that you didn't successfully purchase the said ticket. Something went wrong and your ticket wasn't mined. That ticket hash probably exists only in your local wallet instance and therefore the error when you try to view its details.

**Review by JustinBeBoy — APPROVED** [review]

**Pull request merged**